awx/main/models/events.py

```diff
@@ -535,16 +536,19 @@ def _update_host_summary_from_stats(self, hostnames):
             # in single queries
             host_mapping = dict((summary['host_id'], summary['id']) for summary in JobHostSummary.objects.filter(job_id=job.id).values('id', 'host_id'))
             updated_hosts = set()
+            updated_hosts_list = []
             for h in all_hosts:
                 # if the hostname *shows up* in the playbook_on_stats event
                 if h.name in hostnames:
                     h.last_job_id = job.id
                     updated_hosts.add(h)
+                    updated_hosts.append(h.name)
                 if h.id in host_mapping:
                     h.last_job_host_summary_id = host_mapping[h.id]
                     updated_hosts.add(h)
 
             Host.objects.bulk_update(list(updated_hosts), ['last_job_id', 'last_job_host_summary_id'], batch_size=100)
+            HostMetric.objects.bulk_create([HostMetric(hostname=hostname, last_automation=dt) for hostname in updated_hosts_list], ignore_conflicts=True)
```

Comment:
So here are two things that need to be done: 
--- If the ```hostname``` already exists in the table , update the field ```last_automation```
--- If the ```hostname``` does not already exist in the table, update these two fields ```first_automation``` and ```last_automation```. In this case both of these values ```first_automation``` and ```last_automation``` will be the same. 

Now, I have some queries on that : 
1) Am i supposed to write one query that does these 2 things at once. OR 
2) Am i supposed to write two queries ? One for bulk create and one for bulk update?


Relevant comment : https://github.com/ansible/tower/issues/4687#issuecomment-807378884

Reply:
2 queries. 1 for bulk create 1 for bulk update.

Reply:
Agreed.  I don't believe it's feasible to do it as a single query.  Though I think the 2nd query can and should be just a `.filter(...).update(...)`, with suitable parameters.  With `.bulk_update(...)`, you would have to already have the object instances.

Reply:
Actually, I take that back.  It _is_ possible to do it all as one query.